ibm/resource_ibm_private_dns_glb_monitor.go

```diff
@@ -0,0 +1,520 @@
+			}
+		} else {
+			CreateMonitorOptions.SetAllowInsecure(false)
+		}
+		if headerok {
+			expandedmonitorheaders, err := expandGLBMonitors(monitorheaders, pdnsGlbMonitorHeadersName)
+			if err != nil {
+				return err
+			}
+			CreateMonitorOptions.SetHeadersVar(expandedmonitorheaders)
+		}
+
+	} else {
+		if pathok || expectedcodeok || allowinsecureok || methodok || expectedbodyok || headerok {
+			return fmt.Errorf("Monitor path/expected_codes/expected_body/allow_insecure/method/headers is not supported in type TCP")
+		}
+	}
+	response, detail, err := sess.CreateMonitor(CreateMonitorOptions)
+	if err != nil {
+		return fmt.Errorf("Error creating pdns GLB monitor:%s\n%s", err, detail)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", instanceID, *response.ID))
+	return resourceIBMPrivateDNSGLBMonitorRead(d, meta)
+}
+
+func expandGLBMonitors(header interface{}, geoType string) ([]dnssvcsv1.HealthcheckHeader, error) {
+	headers := header.(*schema.Set).List()
+	expandheaders := make([]dnssvcsv1.HealthcheckHeader, 0)
+	for _, v := range headers {
+		locationConfig := v.(map[string]interface{})
+		hname := locationConfig[pdnsGlbMonitorHeadersName].(string)
+		headers := expandStringList(locationConfig[pdnsGlbMonitorHeadersValue].([]interface{}))
+		headerItem := dnssvcsv1.HealthcheckHeader{
+			Name:  &hname,
+			Value: headers,
+		}
+		expandheaders = append(expandheaders, headerItem)
+	}
+	return expandheaders, nil
+}
+
+func resourceIBMPrivateDNSGLBMonitorRead(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+	idset := strings.Split(d.Id(), "/")
+
+	getMonitorOptions := sess.NewGetMonitorOptions(idset[0], idset[1])
+	response, detail, err := sess.GetMonitor(getMonitorOptions)
+	if err != nil {
+		return fmt.Errorf("Error fetching pdns GLB Monitor:%s\n%s", err, detail)
+	}
+
+	d.Set("id", response.ID)
+	d.Set(pdnsInstanceID, idset[0])
+	d.Set(pdnsGlbMonitorID, response.ID)
+	d.Set(pdnsGlbMonitorName, response.Name)
+	d.Set(pdnsGlbMonitorCreatedOn, response.CreatedOn)
+	d.Set(pdnsGlbMonitorModifiedOn, response.ModifiedOn)
+	d.Set(pdnsGlbMonitorType, response.Type)
+	d.Set(pdnsGlbMonitorPort, response.Port)
+	d.Set(pdnsGlbMonitorInterval, response.Interval)
+	d.Set(pdnsGlbMonitorRetries, response.Retries)
+	d.Set(pdnsGlbMonitorTimeout, response.Timeout)
+	if response.Description != nil {
+		d.Set(pdnsGlbMonitorDescription, response.Description)
+	}
+	if *response.Type == "HTTP" || *response.Type == "HTTPS" {
+		d.Set(pdnsGlbMonitorMethod, response.Method)
+		d.Set(pdnsGlbMonitorPath, response.Path)
+		d.Set(pdnsGlbMonitorExpectedCodes, response.ExpectedCodes)
+		d.Set(pdnsGlbMonitorHeaders, flattenDataSourceLoadBalancerHeader(response.HeadersVar))
+		if response.ExpectedBody != nil {
+			d.Set(pdnsGlbMonitorExpectedBody, response.ExpectedBody)
+		}
+	}
+	if *response.Type == "HTTPS" {
+		if response.AllowInsecure != nil {
```

Comment:
Cant this if be moved to previous if?

Reply:
I believe In the case of tcp it ll be `nil`. So you need not check for types..
Just have a check for nil..
If it is not nil then set else ignore..

Reply:
Handled